### ISSUE TITLE
AWSKinesisVideoDemoApp has stopped

### ISSUE THREAD

**Issue by evi8899** [awslabs/aws-sdk-android-samples#191]
Hi,

When I tried AmazonKinesisVideoDemoApp in HTC sense Android version 5.0.2, it gives this error:

```
02-05 11:40:22.511 6535-6535/com.amazonaws.kinesisvideo.demoapp E/Typeface: SANS_LOC file not found.
02-05 11:40:22.531 6535-6535/com.amazonaws.kinesisvideo.demoapp D/IdentityManager: Initializing Cognito
02-05 11:40:22.581 6535-6535/com.amazonaws.kinesisvideo.demoapp D/CognitoCachingCredentialsProvider: Loading credentials from SharedPreferences
02-05 11:40:22.611 6535-6535/com.amazonaws.kinesisvideo.demoapp D/CognitoCachingCredentialsProvider: Loading credentials from SharedPreferences
02-05 11:40:22.641 6535-6535/com.amazonaws.kinesisvideo.demoapp W/art: Before Android 4.1, method android.graphics.PorterDuffColorFilter android.support.graphics.drawable.VectorDrawableCompat.updateTintFilter(android.graphics.PorterDuffColorFilter, android.content.res.ColorStateList, android.graphics.PorterDuff$Mode) would have incorrectly overridden the package-private method in android.graphics.drawable.Drawable
02-05 11:40:22.641 6535-6535/com.amazonaws.kinesisvideo.demoapp E/Typeface: SANS_LOC file not found.
02-05 11:40:22.651 6535-6535/com.amazonaws.kinesisvideo.demoapp D/IdentityManager: Resume session called.
02-05 11:40:22.651 6535-6555/com.amazonaws.kinesisvideo.demoapp D/IdentityManager: Starting up authentication...
02-05 11:40:22.651 6535-6555/com.amazonaws.kinesisvideo.demoapp D/CognitoUserPoolsSignInProvider: initalizing Cognito User Pools
02-05 11:40:22.691 6535-6555/com.amazonaws.kinesisvideo.demoapp I/CognitoUserPoolsSignInProvider: refreshUserSignInState: Signed in with Cognito.
02-05 11:40:22.691 6535-6555/com.amazonaws.kinesisvideo.demoapp D/IdentityManager: Refreshing credentials with identity provider Amazon Cognito Your User Pools
02-05 11:40:22.691 6535-6555/com.amazonaws.kinesisvideo.demoapp D/IdentityManager: federateWithProvider
02-05 11:40:22.691 6535-6555/com.amazonaws.kinesisvideo.demoapp D/IdentityManager: Initializing Cognito
02-05 11:40:22.701 6535-6535/com.amazonaws.kinesisvideo.demoapp D/Atlas: Validating map...
02-05 11:40:22.711 6535-6555/com.amazonaws.kinesisvideo.demoapp D/CognitoCachingCredentialsProvider: Loading credentials from SharedPreferences
02-05 11:40:22.731 6535-6535/com.amazonaws.kinesisvideo.demoapp D/FindExtension: FindExtension: before mHardwareRenderer.initialize, mSurface.isValid() = true
02-05 11:40:22.731 6535-6557/com.amazonaws.kinesisvideo.demoapp I/Adreno-EGL: <qeglDrvAPI_eglInitialize:410>: EGL 1.4 QUALCOMM build: AU_LINUX_ANDROID_LA.AF.1.1_RB1.05.00.02.006.020 - CR771817 ()
                                                                              OpenGL ES Shader Compiler Version: E031.25.03.06
                                                                              Build Date: 03/04/15 Wed
                                                                              Local Branch: 
                                                                              Remote Branch: refs/tags/AU_LINUX_ANDROID_LA.AF.1.1_RB1.05.00.02.006.020
                                                                              Local Patches: NONE
                                                                              Reconstruct Branch: NOTHING
02-05 11:40:22.751 6535-6555/com.amazonaws.kinesisvideo.demoapp D/CognitoCachingCredentialsProvider: Loading credentials from SharedPreferences
02-05 11:40:22.771 6535-6558/com.amazonaws.kinesisvideo.demoapp D/CognitoCachingCredentialsProvider: Clearing credentials from SharedPreferences
02-05 11:40:22.771 6535-6558/com.amazonaws.kinesisvideo.demoapp D/CognitoCachingCredentialsProvider: Identity id is changed
02-05 11:40:22.771 6535-6558/com.amazonaws.kinesisvideo.demoapp D/CognitoCachingCredentialsProvider: Saving identity id to SharedPreferences
02-05 11:40:22.771 6535-6558/com.amazonaws.kinesisvideo.demoapp D/CognitoCachingCredentialsProvider: Clearing credentials from SharedPreferences
02-05 11:40:22.771 6535-6558/com.amazonaws.kinesisvideo.demoapp D/CognitoCachingCredentialsProvider: Clearing credentials from SharedPreferences
02-05 11:40:22.771 6535-6558/com.amazonaws.kinesisvideo.demoapp D/CognitoCachingCredentialsProvider: Clearing credentials from SharedPreferences
02-05 11:40:22.771 6535-6558/com.amazonaws.kinesisvideo.demoapp D/IdentityManager: refresh credentials
02-05 11:40:22.771 6535-6558/com.amazonaws.kinesisvideo.demoapp D/AWSRefreshingCognitoIdentityProvider: Refreshing token...
02-05 11:40:22.811 6535-6558/com.amazonaws.kinesisvideo.demoapp D/libc: [NET] android_getaddrinfofornet+,hn 40(0x636f676e69746f),sn(),hints(known),family 0,flags 4
02-05 11:40:22.811 6535-6558/com.amazonaws.kinesisvideo.demoapp D/libc: [NET] android_getaddrinfofornet-, err=8
02-05 11:40:22.811 6535-6558/com.amazonaws.kinesisvideo.demoapp D/libc: [NET] android_getaddrinfofornet+,hn 40(0x636f676e69746f),sn(),hints(known),family 0,flags 1024
02-05 11:40:22.811 6535-6558/com.amazonaws.kinesisvideo.demoapp D/libc: [NET] android_getaddrinfofornet-, pass to proxy
02-05 11:40:22.811 6535-6558/com.amazonaws.kinesisvideo.demoapp D/libc: [NET] android_getaddrinfo_proxy+
02-05 11:40:22.811 6535-6558/com.amazonaws.kinesisvideo.demoapp D/libc: [NET] android_getaddrinfo_proxy get netid:0
02-05 11:40:22.841 6535-6558/com.amazonaws.kinesisvideo.demoapp D/libc: [NET] android_getaddrinfo_proxy-, success
02-05 11:40:23.862 6535-6558/com.amazonaws.kinesisvideo.demoapp D/CognitoCachingCredentialsProvider: Identity id is changed
02-05 11:40:23.862 6535-6558/com.amazonaws.kinesisvideo.demoapp D/CognitoCachingCredentialsProvider: Saving identity id to SharedPreferences
02-05 11:40:23.862 6535-6558/com.amazonaws.kinesisvideo.demoapp D/CognitoCachingCredentialsProvider: Clearing credentials from SharedPreferences
02-05 11:40:24.103 6535-6558/com.amazonaws.kinesisvideo.demoapp D/AWSRefreshingCognitoIdentityProvider: Refreshing token...
02-05 11:40:24.673 6535-6558/com.amazonaws.kinesisvideo.demoapp D/CognitoCachingCredentialsProvider: Saving credentials to SharedPreferences
02-05 11:40:24.683 6535-6558/com.amazonaws.kinesisvideo.demoapp D/CognitoCachingCredentialsProvider: Saving identity id to SharedPreferences
02-05 11:40:24.944 6535-6558/com.amazonaws.kinesisvideo.demoapp D/CognitoCachingCredentialsProvider: Saving credentials to SharedPreferences
02-05 11:40:24.994 6535-6558/com.amazonaws.kinesisvideo.demoapp D/IdentityManager: SignInProviderResultAdapter.onCognitoSuccess()
02-05 11:40:24.994 6535-6535/com.amazonaws.kinesisvideo.demoapp D/IdentityManager: Successfully got credentials from identity provider 'Amazon Cognito Your User Pools
02-05 11:40:25.054 6535-6550/com.amazonaws.kinesisvideo.demoapp I/art: Background sticky concurrent mark sweep GC freed 36151(2MB) AllocSpace objects, 12(216KB) LOS objects, 54% free, 1950KB/4MB, paused 8.606ms total 131.970ms
02-05 11:40:25.054 6535-6535/com.amazonaws.kinesisvideo.demoapp E/Typeface: SANS_LOC file not found.
02-05 11:40:25.204 6535-6535/com.amazonaws.kinesisvideo.demoapp I/ListPopupWindow: Could not find method setEpicenterBounds(Rect) on PopupWindow. Oh well.
02-05 11:40:25.224 6535-6535/com.amazonaws.kinesisvideo.demoapp E/art: dlopen(".so", RTLD_LAZY) failed: dlopen failed: library ".so" not found
02-05 11:40:25.224 6535-6535/com.amazonaws.kinesisvideo.demoapp W/KinesisVideoAndroidClient: Unsatisfied link error. Directly loading native library .so.
02-05 11:40:25.224 6535-6535/com.amazonaws.kinesisvideo.demoapp E/art: dlopen(".dylib", RTLD_LAZY) failed: dlopen failed: library ".dylib" not found
02-05 11:40:25.224 6535-6535/com.amazonaws.kinesisvideo.demoapp W/KinesisVideoAndroidClient: Unsatisfied link error. Directly loading native library .dylib.
02-05 11:40:25.224 6535-6535/com.amazonaws.kinesisvideo.demoapp E/art: dlopen(".dll", RTLD_LAZY) failed: dlopen failed: library ".dll" not found
02-05 11:40:25.224 6535-6535/com.amazonaws.kinesisvideo.demoapp W/KinesisVideoAndroidClient: Unsatisfied link error. Directly loading native library .dll.
02-05 11:40:25.224 6535-6535/com.amazonaws.kinesisvideo.demoapp V/KinesisVideoAndroidClient: Success! Directly loaded native library libKinesisVideoProducerJNI.so.
02-05 11:40:25.224 6535-6535/com.amazonaws.kinesisvideo.demoapp V/KinesisVideoAndroidClient: KinesisVideoProducerJNI library: version 1.1, compile time Nov 27 2017 18:10:15
02-05 11:40:25.224 6535-6535/com.amazonaws.kinesisvideo.demoapp V/platform-utils: Java_com_amazonaws_kinesisvideo_producer_jni_NativeKinesisVideoProducerJni_createKinesisVideoClient(): Enter
02-05 11:40:25.224 6535-6535/com.amazonaws.kinesisvideo.demoapp I/platform-utils: Java_com_amazonaws_kinesisvideo_producer_jni_NativeKinesisVideoProducerJni_createKinesisVideoClient(): Creating Kinesis Video client.
02-05 11:40:25.224 6535-6535/com.amazonaws.kinesisvideo.demoapp I/platform-utils: createKinesisVideoClient(): Creating Kinesis Video Client
02-05 11:40:25.224 6535-6535/com.amazonaws.kinesisvideo.demoapp I/platform-utils: heapInitialize(): Initializing native heap with limit size 805306368, spill ratio 90% and flags 0x00000001
02-05 11:40:25.224 6535-6535/com.amazonaws.kinesisvideo.demoapp I/platform-utils: heapInitialize(): Creating AIV heap.
02-05 11:40:25.224 6535-6535/com.amazonaws.kinesisvideo.demoapp E/libc: mmap fail (pid 6535, tid 6535, size 805310464, flags 0x22, errno 12(Out of memory))
02-05 11:40:25.224 6535-6535/com.amazonaws.kinesisvideo.demoapp E/libc: mmap fail (pid 6535, tid 6535, size 805310464, flags 0x22, errno 12(Out of memory))
02-05 11:40:25.224 6535-6535/com.amazonaws.kinesisvideo.demoapp E/platform-utils: aivHeapInit(): Failed to allocate heap with limit size 805306368
02-05 11:40:25.224 6535-6535/com.amazonaws.kinesisvideo.demoapp E/platform-utils: heapInitialize(): Failed to initialize native heap.
02-05 11:40:25.224 6535-6535/com.amazonaws.kinesisvideo.demoapp I/platform-utils: heapRelease(): Freeing native heap.
02-05 11:40:25.224 6535-6535/com.amazonaws.kinesisvideo.demoapp I/platform-utils: freeKinesisVideoClient(): Freeing Kinesis Video Client
02-05 11:40:25.224 6535-6535/com.amazonaws.kinesisvideo.demoapp A/libc: Fatal signal 11 (SIGSEGV), code 1, fault addr 0x0 in tid 6535 (isvideo.demoapp)
02-05 11:40:25.224 6535-6535/com.amazonaws.kinesisvideo.demoapp W/libc: Security Level: (1), Debug inforamtion is controlled by the DUMPABLE flag.
```

Can anyone point out the problem?

Thanks in advance

**Comment by minbi** [comment]
Hi @evi8899 ,

Thanks for reporting this issue. We will investigate and update you when we find a solution.

**Comment by minbi** [comment]
Hi @evi8899 ,

The native heap size allocated is too large and we are working to identify a more proper API to dynamically choose the largest possible heap size.

**Comment by tavoaqp** [comment]
Hi, I also have the same error on Motorola G6 Play with Android 8.0:
`07/16 22:08:41: Launching AmazonKinesisVideoDemoApp
$ adb install-multiple -r -t /home/gtorres/git/aws-sdk-android-samples/AmazonKinesisVideoDemoApp/build/intermediates/resources/instant-run/debug/resources-debug.apk /home/gtorres/git/aws-sdk-android-samples/AmazonKinesisVideoDemoApp/build/intermediates/split-apk/debug/slices/slice_7.apk /home/gtorres/git/aws-sdk-android-samples/AmazonKinesisVideoDemoApp/build/intermediates/split-apk/debug/slices/slice_6.apk /home/gtorres/git/aws-sdk-android-samples/AmazonKinesisVideoDemoApp/build/intermediates/split-apk/debug/slices/slice_8.apk /home/gtorres/git/aws-sdk-android-samples/AmazonKinesisVideoDemoApp/build/intermediates/split-apk/debug/dep/dependencies.apk /home/gtorres/git/aws-sdk-android-samples/AmazonKinesisVideoDemoApp/build/intermediates/split-apk/debug/slices/slice_9.apk /home/gtorres/git/aws-sdk-android-samples/AmazonKinesisVideoDemoApp/build/intermediates/split-apk/debug/slices/slice_4.apk /home/gtorres/git/aws-sdk-android-samples/AmazonKinesisVideoDemoApp/build/intermediates/split-apk/debug/slices/slice_2.apk /home/gtorres/git/aws-sdk-android-samples/AmazonKinesisVideoDemoApp/build/intermediates/split-apk/debug/slices/slice_5.apk /home/gtorres/git/aws-sdk-android-samples/AmazonKinesisVideoDemoApp/build/intermediates/split-apk/debug/slices/slice_0.apk /home/gtorres/git/aws-sdk-android-samples/AmazonKinesisVideoDemoApp/build/intermediates/split-apk/debug/slices/slice_3.apk /home/gtorres/git/aws-sdk-android-samples/AmazonKinesisVideoDemoApp/build/intermediates/split-apk/debug/slices/slice_1.apk /home/gtorres/git/aws-sdk-android-samples/AmazonKinesisVideoDemoApp/build/intermediates/instant-run-apk/debug/AmazonKinesisVideoDemoApp-debug.apk 
Split APKs installed
$ adb shell am start -n "com.amazonaws.kinesisvideo.demoapp/com.amazonaws.kinesisvideo.demoapp.activity.StartUpActivity" -a android.intent.action.MAIN -c android.intent.category.LAUNCHER
Client not ready yet..Waiting for process to come online
Connected to process 23301 on device motorola-moto_g_6__play-0046844023
Capturing and displaying logcat messages from application. This behavior can be disabled in the "Logcat output" section of the "Debugger" settings page.
W/ActivityThread: Application com.amazonaws.kinesisvideo.demoapp can be debugged on port 8100...
I/InstantRun: starting instant run server: is main process
D/IdentityManager: Creating the Cognito Caching Credentials Provider with a refreshing Cognito Identity Provider.
D/CognitoCachingCredentialsProvider: Loading credentials from SharedPreferences
D/IdentityManager: Resume Session called.
D/IdentityManager: Looking for a previously signed-in session.
D/CognitoUserPoolsSignInProvider: initalizing Cognito User Pools
D/SignInManager: Providers: [{class com.amazonaws.mobile.auth.userpools.CognitoUserPoolsSignInProvider=com.amazonaws.mobile.auth.userpools.CognitoUserPoolsSignInProvider@d8f9e9b}]
D/OpenGLRenderer: HWUI GL Pipeline
I/Adreno: QUALCOMM build                   : 3273d9c, I0b59f3a7cf
          Build Date                       : 02/27/18
          OpenGL ES Shader Compiler Version: EV031.22.00.01
          Local Branch                     : 
          Remote Branch                    : refs/tags/AU_LINUX_ANDROID_LA.UM.6.5.R1.08.00.00.312.044
          Remote Branch                    : NONE
          Reconstruct Branch               : NOTHING
I/CognitoUserPoolsSignInProvider: refreshUserSignInState: Signed in with Cognito.
D/SignInManager: Refreshing provider: Amazon Cognito Your User Pools
D/IdentityManager: Refreshing credentials with sign-in provider Amazon Cognito Your User Pools
D/IdentityManager: federate with provider: Populate loginsMap with token.
D/CognitoCachingCredentialsProvider: Clearing credentials from SharedPreferences
D/CognitoCachingCredentialsProvider: Identity id is changed
                                     Saving identity id to SharedPreferences
D/CognitoCachingCredentialsProvider: Clearing credentials from SharedPreferences
I/Adreno: PFP: 0x005ff087, ME: 0x005ff063
D/CognitoCachingCredentialsProvider: Clearing credentials from SharedPreferences
                                     Clearing credentials from SharedPreferences
D/IdentityManager: refresh credentials
D/AWSRefreshingCognitoIdentityProvider: Storing the Refresh token in the loginsMap.
I/OpenGLRenderer: Initialized EGL, version 1.4
D/OpenGLRenderer: Swap behavior 2
D/NetworkSecurityConfig: No Network Security Config specified, using platform default
D/CognitoCachingCredentialsProvider: Identity id is changed
                                     Saving identity id to SharedPreferences
                                     Clearing credentials from SharedPreferences
D/AWSRefreshingCognitoIdentityProvider: Storing the Refresh token in the loginsMap.
D/CognitoCachingCredentialsProvider: Saving credentials to SharedPreferences
D/CognitoCachingCredentialsProvider: Saving identity id to SharedPreferences
D/CognitoCachingCredentialsProvider: Saving credentials to SharedPreferences
D/IdentityManager: SignInProviderResultAdapter.onCognitoSuccess()
D/IdentityManager: Successfully got AWS Credentials.
I/zygote: Do partial code cache collection, code=30KB, data=26KB
I/zygote: After code cache collection, code=30KB, data=26KB
          Increasing code cache capacity to 128KB
W/KinesisVideoAndroidClient: Unsatisfied link error. Directly loading native library .so.
W/KinesisVideoAndroidClient: Unsatisfied link error. Directly loading native library .dylib.
W/KinesisVideoAndroidClient: Unsatisfied link error. Directly loading native library .dll.
                             Unsatisfied link error. Directly loading native library libKinesisVideoProducerJNI.so.
W/KinesisVideoAndroidClient: Unsatisfied link error. Directly loading native library libKinesisVideoProducerJNI.dylib.
W/KinesisVideoAndroidClient: Unsatisfied link error. Directly loading native library libKinesisVideoProducerJNI.dll.
V/KinesisVideoAndroidClient: Success! Indirectly loaded native library KinesisVideoProducerJNI.
V/KinesisVideoAndroidClient: KinesisVideoProducerJNI library: version 1.5, compile time Mar 29 2018 00:49:24
V/platform-utils: Java_com_amazonaws_kinesisvideo_producer_jni_NativeKinesisVideoProducerJni_createKinesisVideoClient(): Enter
I/platform-utils: Java_com_amazonaws_kinesisvideo_producer_jni_NativeKinesisVideoProducerJni_createKinesisVideoClient(): Creating Kinesis Video client.
                  createKinesisVideoClient(): Creating Kinesis Video Client
                  heapInitialize(): Initializing native heap with limit size 805306368, spill ratio 90% and flags 0x00000001
                  heapInitialize(): Creating AIV heap.
E/platform-utils: aivHeapInit(): Failed to allocate heap with limit size 805306368
                  heapInitialize(): Failed to initialize native heap.
I/platform-utils: heapRelease(): Freeing native heap.
                  freeKinesisVideoClient(): Freeing Kinesis Video Client
A/libc: Fatal signal 11 (SIGSEGV), code 1, fault addr 0x0 in tid 23301 (isvideo.demoapp)
Application terminated.`

**Comment by muellerfr** [comment]
Hello @evi8899,

We released a fix with v2.8.0 of the SDK. Please upgrade and let us know if you are still seeing the error.

**Comment by muellerfr** [comment]
This issue has been closed because of inactivity. Please open a new issue in https://github.com/aws/aws-sdk-android/issues if you are still encountering problems.
